logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/sso/SSOInitiateLoginUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.wire.kalium.logic.feature.auth.sso
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.ServerConfig
+import com.wire.kalium.logic.data.auth.login.SSOLoginRepository
+import com.wire.kalium.logic.feature.auth.ValidateUUIDUseCase
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.network.exceptions.KaliumException
+import io.ktor.http.HttpStatusCode
+
+sealed class SSOInitiateLoginResult {
+    data class Success(val requestUrl: String) : SSOInitiateLoginResult()
+
+    sealed class Failure : SSOInitiateLoginResult() {
+        object InvalidCode : Failure()
+        object InvalidRedirect : Failure()
+        class Generic(val genericFailure: CoreFailure) : Failure()
+    }
+}
+
+data class SSORedirects(val success: String, val error: String)
+
+interface SSOInitiateLoginUseCase {
+    sealed class Param {
+        abstract val ssoCode: String
+        abstract val serverConfig: ServerConfig
+        data class NoRedirect(override val ssoCode: String, override val serverConfig: ServerConfig): Param()
```

Comment:
What do you think of `WithRedirect` and `WithoutRedirect`?

Reply:
Sure, that's better, I changed it